From 3ca3c90ed1d5a569724cb4ca54c8fb6911e3a3c0 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Ren=C3=A9=20Stadler?= <rene.stadler@nokia.com>
Date: Tue, 24 Feb 2009 02:26:30 +0200
Subject: [PATCH] aacparse: Bump rank to PRIMARY+1

---
 gst/aacparse/gstaacparse.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/gst/aacparse/gstaacparse.c b/gst/aacparse/gstaacparse.c
index 3f15a62..ce107df 100644
--- a/gst/aacparse/gstaacparse.c
+++ b/gst/aacparse/gstaacparse.c
@@ -872,7 +872,7 @@ static gboolean
 plugin_init (GstPlugin * plugin)
 {
   return gst_element_register (plugin, "aacparse",
-      GST_RANK_NONE, GST_TYPE_AACPARSE);
+      GST_RANK_PRIMARY + 1, GST_TYPE_AACPARSE);
 }
 
 
